openedx/features/ucsd_features/utils.py

```diff
@@ -1,46 +1,61 @@
 import json
 import logging
-from smtplib import SMTPException
 
 from django.conf import settings
-from django.core.mail import EmailMultiAlternatives, send_mail
-from django.template.loader import get_template
+from edx_ace import ace
+from edx_ace.recipient import Recipient
+from openedx.features.ucsd_features.message_types import SupportNotification
+from openedx.core.djangoapps.theming.helpers import get_current_site
+from openedx.core.djangoapps.ace_common.template_context import get_base_template_context
 
 log = logging.getLogger(__name__)
 TEMPLATE_PATH = '{key}_email.html'
 
 
-def send_notification(key, data, subject, from_email, dest_emails):
+def send_notification(message_type, data, subject, dest_emails):
     """
-    Send an email.
+    Send an email
 
-    params:
-        key - Email template will be selected on the basis of key
+    Arguments:
+        message_type - string value to select ace message object
         data - Dict containing context/data for the template
         subject - Email subject
-        from_email - Email address to send email
         dest_emails - List of destination emails
 
-    return: a boolean variable indicating email response.
+    Returns:
+        a boolean variable indicating email response.
     """
+    message_types = {
+        'support': SupportNotification,
+    }
+    current_site = get_current_site()
     content = json.dumps(data)
-    email_template_path = TEMPLATE_PATH.format(key=key)
-    html_content = get_template(email_template_path).render(data)
-    msg = EmailMultiAlternatives(subject, content, from_email, dest_emails)
-    msg.attach_alternative(html_content, "text/html")
+    data.update(
+        {
+            'subject': subject,
+            'site': current_site
+        }
+    )
+    message_context = get_base_template_context(current_site)
+    message_context.update(data)
+    message_class = message_types[message_type]
     try:
-        response = msg.send()
-        log.info(
-            'Email has been sent from "%s" to "%s" for content %s.',
-            from_email,
-            dest_emails,
-            content
-        )
-        return response
-    except SMTPException:
+        for email in dest_emails:
+            message = message_class().personalize(
+                recipient=Recipient(username='', email_address=email),
+                language='en',
+                user_context=message_context,
+            )
+            ace.send(message)
+            log.info(
+                'Email has been sent to "%s" for content %s.',
+                dest_emails,
```

Comment:
why we are looping this `dest_emails`? we are logging the same value `dest_emails` in every iteration.

Reply:
changes have been made.